external-providers/generic-external-provider/main.go

```diff
@@ -35,4 +35,22 @@ func main() {
 	s := provider.NewServer(client, *port, log)
 	ctx := context.TODO()
 	s.Start(ctx)
+	// serviceClient, err := client.Init(ctx, log, provider.InitConfig{
```

Comment:
We should remove this, but being able to run the provider directly has been really nice for debugging. Maybe should add a debug subcommand or something for this.

Reply:
make it a hidden subcommand in the main binary?

Reply:
I don't think it can work in the main binary, would have to be in the provider binaries since the communication between the two is over RPC. Here you can delve debug / see the print statements.

Side note, we really should be capturing the stdout from these providers and presenting them to the users, I don't think we do currently (right?)

Reply:
Not in a real way, There is the plumbing to do it though

We have one set of questions with that, though,

1. Do we print it out as it happens, or do we bunch them together once a given evaluation is finished
2. Are there log levels that need to be returned, is this a verbosity thing where we only print on high levels?

Let me know if you want me to take a stab at it. @pranavgaikwad I am out of the loop on what is important now.

Reply:
I think we should print them as they happen, I wouldn't necessarily bother with verbosity levels as a first-class thing since a provider can always implement verbosity levels through `providerSpecificConfig`. So we'd just want to log with an identifier that lets you know which provider name is logging and do it in real time so that hangs/etc can be debugged

Reply:
I think if you wanted to take a stab at it it would be a huge boon to developing providers. such a PITA right now

Reply:
@shawn-hurley @fabianvf let me know if this sums up the work https://github.com/konveyor/analyzer-lsp/issues/312

Reply:
will do then!